viper_test.go

```diff
@@ -1436,24 +1307,203 @@ hobbies:
 name: steve
 `)
 
-func TestWriteConfigYAML(t *testing.T) {
-	v := New()
+func TestWriteConfig(t *testing.T) {
 	fs := afero.NewMemMapFs()
-	v.SetFs(fs)
-	v.SetConfigName("c")
-	v.SetConfigType("yaml")
-	err := v.ReadConfig(bytes.NewBuffer(yamlExample))
-	if err != nil {
-		t.Fatal(err)
+	testCases := map[string]struct {
```

Comment:
Nice refactor!

Currently, the config type is always inferred from the file name, not the original config type. So if the config type was yaml, but I provided a json file name, the output content was in JSON.

Can you please add a test case that verifies this is still the case?

Because (unfortunately) as far as I can tell, it is not.

Reply:
Just making sure I got the right idea here, are you saying to default on `writeConfig` to the extension file and if this is blank then use the setting for config type?

Reply:
Yup. At least, that's compatible with the current behaviour, taking the original configType by default is not.

Reply:
Gotcha 👍